hepdata/modules/records/views.py

```diff
@@ -610,23 +616,48 @@ def get_resource(resource_id):
 
 
 @blueprint.route('/<int:recid>/consume', methods=['GET', 'POST'])
-@login_required
 def consume_data_payload(recid):
     """
     This method persists, then presents the loaded data back to the user.
+    It is also used by the hepdata-cli tool to upload a submission.
 
     :param recid: record id to attach the data to
     :return: page rendering
     """
 
-    if request.method == 'POST':
-        file = request.files['hep_archive']
-        redirect_url = request.url_root + "record/{}"
-        return process_payload(recid, file, redirect_url)
-
-    else:
+    if request.method == 'GET':
         return redirect('/record/' + str(recid))
 
+    if current_user.is_authenticated is False:  # code for hepdata-cli upload
```

Comment:
Yes, you get a bed request (I tried this on the local website by going to the sandbox upload page, and then restarting the local website to timeout the login). What if in the upload request from hepdata-cli I hide another variable (besides email, invitation cookie, etc..) which tags the fact that the request comes from the terminal. Then if the variable is there you get bad request error, if it is not it means you are on the browser and get redirected. Do you think this could work?

I'm not sure about the authentication part. Should I look into how to login someone from the terminal? I think with Graeme we thought that would be much more complicated than just having an invitation cookie.

Reply:
Added:
```from flask_security.utils import verify_password```
and
```verify_password(request.form['pswd'], user.password)```